Test_oM/ITestInformation.cs

```diff
@@ -20,22 +20,26 @@
  * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
  */
 
-using System;
+using BH.oM.Base;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Collections.ObjectModel;
 using System.ComponentModel;
+using System.Linq;
+using System;
 
-namespace BH.oM.Test.Results
+namespace BH.oM.Test
 {
-    [Description("Outcome of a test (e.g. Pass or Fail).")]
-    public enum ResultStatus
+    public interface ITestInformation : IObject
     {
-        Undefined,
-        Pass,
-        Fail,
-        CriticalFail,
-    }
-}
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        string Message { get; }
+        TestInformationStatus Status { get; set; }
```

Comment:
Do not think this should be set-able on the level on the interface?

If some class that implements this wants to add a setter, that is ofc fine, but on the base interface I would keep this get only, especially as it is forcing a setter onto the `TestResult` if we have a setter here.

```suggestion
        TestInformationStatus Status { get; }
```

Reply:
Because we are defaulting to an `Error` status I opted to make it easy for people to change this status as this might change far more easily as a result of examination than the other properties.

Reply:
For some implementation for some particular class that implements this interface, I get your point, but as I stated above, there is nothing stopping an implementer to add a setter for that case.

This is forcing you to add one though, and making this settable almost to me makes the point of making the `TestResult` class immutable mute, as this fundamentally is the result itself. To me it feels like saying that the absolute key bit of information is adjustable, while the adjoining bit is not, which I am not sure I really agree with.

Reply:
Ah ok I misunderstood, I thought you were advocating for removing it completely. See latest comment, removed `set` from interface.